hammers/scripts/k8s_calico_ip_cleanup.py

```diff
@@ -0,0 +1,104 @@
+'''
+Basic Usage:
+
+.. code-block:: bash
+
+    k8s-calico-ip-cleanup --kubeconfig_path <path_to_site_kubeconfig>
+
+This script cleans up the calico ipv4 address annotation for offline nodes in a kubernetes cluster.
+Cleaning up these addresses is necessary to allow new nodes to start with the same ip addresses.
+'''
+
+import os
+import sys
+
+from kubernetes import client, config
+
+from hammers.util import base_parser
+
+# Kubernetes 10.x/12.x support
+try:
+    K8sApiException = client.ApiException  # >=12.x
+except:
+    K8sApiException = client.api_client.ApiException
+
+_KUBERNETES_CLIENT = None
+
+_KUBECONFIG_PATH = None
+
+def kubernetes_client():
+    global _KUBERNETES_CLIENT
+    if not _KUBERNETES_CLIENT:
+        config.load_kube_config(config_file=_KUBECONFIG_PATH)
+        _KUBERNETES_CLIENT = client.CoreV1Api()
+    return _KUBERNETES_CLIENT
+
+
+def clear_calico_ipv4_annotation(node_name):
+    try:
+        v1 = kubernetes_client()
+
+        node = v1.read_node(name=node_name)
+
+        # Remove the Calico IPv4 annotation
+        if "projectcalico.org/IPv4Address" in node.metadata.annotations:
+            del node.metadata.annotations["projectcalico.org/IPv4Address"]
+
+            v1.replace_node(name=node_name, body=node)
+            print(f"Cleared Calico IPv4 annotation for node: {node_name}")
+
+    except Exception as e:
+        print(f"Error clearing Calico IPv4 annotation for node {node_name}: {e}")
+
+def get_offline_nodes():
+    try:
+        v1 = kubernetes_client()
+
+        nodes = v1.list_node(watch=False)
+
+        # Filter nodes based on their status (considered offline if not ready)
+        offline_nodes = [
+            node.metadata.name
+            for node in nodes.items
+            if node.status.conditions is None
+            or all(
+                cond.status != "True"
+                for cond in node.status.conditions
+                if cond.type == "Ready"
+            )
+        ]
```

Comment:
There are multiple reasons a node may not be `ready`, including that calico is still starting.

we should be more specific, such as:
> `node.kubernetes.io/unreachable`: Node is unreachable from the node controller. This corresponds to the `NodeCondition` `Ready` being `Unknown`.

see https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions

Reply:
right, on it